instrumentation/net/http/otelhttp/transport.go

```diff
@@ -109,21 +143,69 @@ func (t *Transport) RoundTrip(r *http.Request) (*http.Response, error) {
 		ctx = httptrace.WithClientTrace(ctx, t.clientTrace(ctx))
 	}
 
+	readRecordFunc := func(int64) {}
+	if t.readEvent {
+		readRecordFunc = func(n int64) {
+			span.AddEvent("read", trace.WithAttributes(ReadBytesKey.Int64(n)))
+		}
+	}
+
+	var bw bodyWrapper
+	// if request body is nil or NoBody, we don't want to mutate the body as it
+	// will affect the identity of it in an unforeseeable way because we assert
+	// ReadCloser fulfills a certain interface and it is indeed nil or NoBody.
+	if r.Body != nil && r.Body != http.NoBody {
+		bw.ReadCloser = r.Body
+		bw.record = readRecordFunc
+		r.Body = &bw
+	}
+
 	r = r.WithContext(ctx)
 	span.SetAttributes(httpconv.ClientRequest(r)...)
+	if t.getRequestAttributes != nil {
+		span.SetAttributes(t.getRequestAttributes(r)...)
+	}
 	t.propagators.Inject(ctx, propagation.HeaderCarrier(r.Header))
 
 	res, err := t.rt.RoundTrip(r)
 	if err != nil {
 		span.RecordError(err)
 		span.SetStatus(codes.Error, err.Error())
 		span.End()
-		return res, err
+	} else {
+		span.SetAttributes(httpconv.ClientResponse(res)...)
+		if t.getResponseAttributes != nil {
+			span.SetAttributes(t.getResponseAttributes(res)...)
+		}
+		span.SetStatus(httpconv.ClientStatus(res.StatusCode))
+		res.Body = newWrappedBody(span, res.Body)
+	}
+
+	// Add metrics
+	attributes := httpconv.ClientRequest(r)
```

Comment:
Yes, we really needed an easy way to filter what attributes we don't want, I can't do any decision at this point in the code.

Reply:
I found in another issue a [view being used to filter attributes](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/3071#issuecomment-1419366500), maybe this could be used here?